### ISSUE TITLE
Batched change notifies child component before parent stache binding

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/can-stache-bindings#278]
@dylanrtt commented on [Wed Feb 10 2016](https://github.com/canjs/canjs/issues/2252)

This bug was introduced in 2.3 (absent in 2.2.9), and is still present in 2.3.15.

``` html
{{#eq page 'view'}}
  <foo-view page="{page}"></foo-view>
{{/eq}}
```

If `<foo-view>` binds to `page`, and `page` is changed in a batch, it seems the component is notified of the change before the stache binding. The stache binding _should_ be called back first to remove the component and hopefully prevent the other batched event from running (however that works).

http://jsbin.com/xeqosawako/edit?html,js,console,output

---

@justinbmeyer commented on [Tue Apr 26 2016](https://github.com/canjs/canjs/issues/2252#issuecomment-214766627)

There's an issue keeping this for 3.0.  Essentially, live-binding relied on a synchronous notification of `removed` events to teardown the bindings.  

In 3.0, for performance reasons, I was hoping to use MutationObservers to know when elements are added or removed.  MutationObservers dispatch at the next turn.

I'm going to try to come up with a way for this to work without having to resort to synchronous `"removed"` events.

---

@justinbmeyer commented on [Tue Apr 26 2016](https://github.com/canjs/canjs/issues/2252#issuecomment-214767961)

actually, listening to nodeLists teardown should work.

**Comment by justinbmeyer** [comment]
I've just run into this.  I'm trying to fix it.  One problem is that the DOM_UI queue is not cancelable.  I could hack around this, but this is something that @chasenlehara should consider for the new `can-bind`.

**Comment by chasenlehara** [comment]
@justinbmeyer Is this fixed in can-stache-bindings 4.2.4?

**Comment by justinbmeyer** [comment]
I believe so
